task/upload: Allow asset IPFS metadata player template on creation

victorges opened the pull request:
We had a legacy check on the export task to only allow assets with playback URLs to 
use player NFT metadata. This was because our very first assets didn't even have an HLS
playlist and just provided MP4 URLs.

For new assets creation, the only other place where `saveNFTMetadata` is called, this is not
necessary tho, as we will always create an HLS playback URL for the asset, or at least have it
playable on the player.

This fixes API-36

victorges committed:
task/export: Move playbackURL check out of common helper

Only the export task should check it.

victorges committed:
Whitespace

0xcadams commented on task/export.go:
```diff
@@ -98,11 +98,16 @@ func uploadFile(tctx *TaskContext, asset *api.Asset, content io.Reader) (*data.E
 	}
 	template, nftMetadata := params.IPFS.NFTMetadataTemplate, params.IPFS.NFTMetadata
+	if template == api.NFTMetadataTemplatePlayer && asset.PlaybackURL == "" {
+		return nil, fmt.Errorf("cannot create player NFT for asset without playback URL")
```
So does this disable the `player` template on create?

victorges replied:
No! For the `upload` task which is what we do on creation, we don't have that check. It should create a player metadata on creation!

0xcadams replied:
Great!

victorges committed:
task/upload: Ignore IPFS storage with null spec

A bit weird interface tbh but it's like that in the API
today and I don't wanna mess with it.
